From 610cf4d8dc1e23d70c83701de40782a15106572a Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 7 May 2020 11:14:15 -0500
Subject: [PATCH 28/31] arm64: dts: beacon-imx8mm: Add variant with spidev
 functions

There are two available SPI buses.  One of them shares pins with
UART3, and the other is routed to a SPI Flash on the baseboard.

However, for people who wantt to manually access the SPI bus with
userspace tools, neither of these buses is available.

This patch adds an alternative device tree with spidev support for
ecah SPI bus.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../dts/freescale/beacon-imx8mm-spidev.dts    | 62 +++++++++++++++++++
 1 file changed, 62 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts
new file mode 100644
index 000000000000..46751be0ac4f
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright 2020 Beacon EmbeddedWorks
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8mm.dtsi"
+#include "beacon-imx8mm-som.dtsi"
+#include "beacon-imx8mm-baseboard.dtsi"
+#include "beacon-imx8mm-lvds.dtsi"
+
+/ {
+	model = "beacon EmbeddedWorks i.MX8M mini Developent Kit";
+	compatible = "fsl,imx8mm";
+
+	chosen {
+		stdout-path = &uart2;
+	};
+
+	aliases {
+		/delete-property/ serial2;
+	};
+};
+
+/* UART3 shares pins with ECSPI1, so they cannot both exist */
+/delete-node/ &uart3;
+
+&ecspi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	status = "okay";
+	cs-gpios = <&gpio5 9 0>;
+	spidev1@0x00 {
+               compatible = "spidev";
+               spi-max-frequency = <5000000>;
+               reg = <0>;
+       };
+};
+
+&ecspi2 {
+	cs-gpios = <&gpio5 13 0>;
+	
+	/delete-node/ at25@0;
+	
+	spidev1@0x00 {
+               compatible = "spidev";
+               spi-max-frequency = <5000000>;
+               reg = <0>;
+       };
+};
+
+&iomuxc {
+	pinctrl_ecspi1: uart3grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_ECSPI1_SCLK_ECSPI1_SCLK		0x82
+			MX8MM_IOMUXC_ECSPI1_MOSI_ECSPI1_MOSI		0x82
+			MX8MM_IOMUXC_ECSPI1_MISO_ECSPI1_MISO		0x82
+			MX8MM_IOMUXC_ECSPI1_SS0_ECSPI1_SS0		0x82
+		>;
+	};
+};
-- 
2.17.1

